### PR TITLE
For #3554 - Add DiffUtil to HistoryAdapter

### PR DESCRIPTION
**Pull request by ekager**
# Pull Request checklist

- [X] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/fenix/blob/master/CHANGELOG.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/library/history/HistoryAdapter.kt [hunk 1]
```diff
@@ -101,11 +102,48 @@ class HistoryAdapter(
     var selected = listOf<HistoryItem>()
 
     fun updateData(items: List<HistoryItem>, mode: HistoryState.Mode) {
+        val diffUtil = DiffUtil.calculateDiff(
```
Comment (colintheshots) [inline app/src/main/java/org/mozilla/fenix/library/history/HistoryAdapter.kt]: Technically, this call should end up on a background thread. This was one of the findings of the performance audit. But it's okay if we make that change later.

File: app/src/main/res/layout/delete_history_button.xml [hunk 2]
```diff
@@ -2,12 +2,16 @@
 <!-- This Source Code Form is subject to the terms of the Mozilla Public
    - License, v. 2.0. If a copy of the MPL was not distributed with this
    - file, You can obtain one at http://mozilla.org/MPL/2.0/. -->
-<com.google.android.material.button.MaterialButton xmlns:android="http://schemas.android.com/apk/res/android"
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (colintheshots) [inline app/src/main/res/layout/delete_history_button.xml]: Why do we need a new layout level surrounding the button? Adding deeper layouts costs a bit of performance.

Comment (ekager) [reply app/src/main/res/layout/delete_history_button.xml]: It has to do with adding an alpha to a recyclerview child. If it isn't nested, the built-in recylerview animations prevent us from adding our own alpha on the button (used for disabled state).

Comment (colintheshots) [reply app/src/main/res/layout/delete_history_button.xml]: Oh, good to know. Thank you.

**Pull request merged**